chingu-voyages/v51-tier3-team-33
# Gary/feature/delete group

# Description
This commit adds the functionality to delete groups. - This is built on top of update-delete-expenses.

## Related to the following Jira/Trello issue 
N/A

## Main changes explained:
- Created deleteGroup route to handle deletion of a group and all records (expenses, usergroups, userexpenses) relating to it.

## How to test:

1. check into current branch
2. do `pnpm install` and `pnpm run dev` to run this PR locally
3. Open postman and create a DELETE request to http://localhost:3000/api/groups/*group id here*/deleteGroup. Choose a group that has atleast 1 expense.
4. The message returned from the API should signal that the group has been deleted successfully. To confirm this, check the following.
5. The group is no longer present in the group collection. 
6. The expense(s) associated with it are no longer present in the expenses collection.
7. The expense_id(s) can no longer be found in the userExpenses collection, 
8. The group id can no longer be found in the user_group collection.
9. If all 4 of those are true. All traces are deleted from the database.


## Note:
Include any information the reviewers need to know.
